spec/models/payment_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Payment, type: :model do
+  it { should belong_to(:site) }
+  it { should belong_to(:agency) }
+
```

Comment:
Need the not null validations, and if possible, the greater-than-zero validation